src/test/java/uk/gov/companieshouse/company/profile/service/CompanyProfileServiceTest.java

```diff
@@ -63,4 +71,34 @@ void getNoCompanyProfileReturned() {
         assertTrue(companyProfileActual.isEmpty());
         verify(logger, times(2)).trace(anyString());
     }
+
+
+    @Test
+    void when_insolvency_data_is_given_then_data_should_be_saved() {
+        CompanyProfile companyProfile = mockCompanyProfileWithoutInsolvency();
+        CompanyProfile companyProfileWithInsolvency = companyProfile;
+        companyProfileWithInsolvency.getData().getLinks().setInsolvency("INSOLVENCY_LINK");
+        when(companyProfileRepository.findByCompanyNumber(Mockito.any())).thenReturn(new CompanyProfileDao(companyProfile));
```

Comment:
Reuse existing query:
`when(companyProfileRepository.findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber(anyString()))
                .thenReturn(Optional.of(new CompanyProfileDao(companyProfile));`